test-utils/src/controller/lst_oracle.rs

```diff
@@ -0,0 +1,78 @@
+use near_sdk::{
+    serde_json::{self, json},
+    AccountId, Gas, NearToken,
+};
+use near_workspaces::{result::ExecutionSuccess, Account, Contract};
+use templar_common::oracle::{
+    price_transformer::PriceTransformer,
+    pyth::{OracleResponse, PriceIdentifier},
+};
+use tokio::sync::OnceCell;
+
+use crate::{define, get_contract};
+
+use super::ContractController;
+
+pub struct LstOracleController {
+    pub contract: Contract,
+}
+
+impl ContractController for LstOracleController {
+    fn contract(&self) -> &Contract {
+        &self.contract
+    }
+}
+
+impl LstOracleController {
+    pub async fn deploy(account: Account, oracle_id: &AccountId) -> Self {
+        static WASM: OnceCell<Vec<u8>> = OnceCell::const_new();
+
+        let wasm = WASM
+            .get_or_init(|| get_contract("templar_lst_oracle_contract", "contract/lst-oracle"))
+            .await;
+
+        let contract = account.deploy(wasm).await.unwrap().unwrap();
+        contract
+            .call("new")
+            .args_json(json!({
+                "oracle_id": oracle_id,
+            }))
+            .transact()
+            .await
+            .unwrap()
+            .unwrap();
```

Comment:
Transactions can fail for a number of reasons on deployment, its best to have it captured what failed.